cli/module_build.go

```diff
@@ -490,6 +491,68 @@ type reloadModuleArgs struct {
 	NoBuild     bool
 	Local       bool
 	NoProgress  bool
+	CloudBuild  bool
+	BuildID     string
+	CloudConfig string
+	ModelName   string
+	Ref         string
+	Token       string
+	Workdir     string
+}
+
+func (c *viamClient) moduleCloudReload(ctx *cli.Context, args reloadModuleArgs, platform string) (string, error) {
+	manifest, err := loadManifest(args.Module)
+	if err != nil {
+		return "", err
+	}
+
+	id := ctx.String(generalFlagID)
+	if id == "" {
+		id = manifest.ModuleID
+	}
+	// download an existing build at the given ID
+	if args.BuildID != "" {
+		downloadArgs := downloadModuleFlags{
+			ID:       id,
+			Version:  args.BuildID,
+			Platform: platform,
+		}
+		return c.downloadModuleAction(ctx, downloadArgs)
+	}
+
+	// create a new build
+	infof(c.c.App.Writer, "Creating a new cloud build and swapping it onto the requested machine part. This may take a few minutes...")
+	buildArgs := moduleBuildStartArgs{
+		Module:    args.Module,
+		Version:   "reload",
+		Ref:       args.Ref,
+		Token:     args.Token,
+		Workdir:   args.Workdir,
+		Platforms: []string{platform},
+	}
+
+	buildID, err := c.moduleBuildStartAction(ctx, buildArgs)
+	if err != nil {
+		return "", err
+	}
+
+	// ensure the build completes before we try to dowload and use it
+	_, err = c.waitForBuildToFinish(buildID, platform)
+	if err != nil {
+		return "", err
+	}
+
+	// There seems to be some delay between a build finishing and being findable for download. In testing,
```

Comment:
can you ticket this ('cloud build wait requires an additional sleep' or something) + drop the ticket here? this is probably a me problem

Reply:
[done](https://viam.atlassian.net/browse/RSDK-11517)